### PR TITLE
action: get latest tag by github api

### PR DESCRIPTION
**Pull request by Desiki-high**
## Details
Use [Github API](https://api.github.com/repos/Dragonflyoss/nydus/releases/latest) to get the latest tag of nydus, and used in smoke/integration-test.

Enhancement  https://github.com/dragonflyoss/nydus/pull/1462.

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by Desiki-high** [comment]
If the `latest` release is stable, I think this patch can work batter. cc@imeoer.

File: .github/workflows/smoke.yml [hunk 1]
```diff
@@ -177,8 +177,11 @@ jobs:
         sudo install -D -m 755 contrib/nydusify/cmd/nydusify /usr/bin/nydus-latest
         sudo install -D -m 755 target/release/nydusd target/release/nydus-image /usr/bin/nydus-latest
 
-        versions=(v0.1.0 ${{env.NYDUS_STABLE_VERSION}} latest)
-        version_exports=(v0_1_0 ${{env.NYDUS_STABLE_VERSION_EXPORT}} latest)
+        export NYDUS_STABLE_VERSION=$(curl https://api.github.com/repos/Dragonflyoss/nydus/releases/latest | jq -r '.tag_name')
```
Comment (imeoer) [inline .github/workflows/smoke.yml:180]: Maybe can reuse the env defined above.

**Review by imeoer — COMMENTED** [review]
others LGTM, thanks!

File: .github/workflows/smoke.yml (hunk 1 above)
Comment (Desiki-high) [reply .github/workflows/smoke.yml:180]: It seems good.

**Comment by Desiki-high** [comment]
In `Github Action`, the environment variable can't share between different steps in the same job, see https://github.com/dragonflyoss/nydus/actions/runs/6766960684/job/18388873567. So we need to export 
`NYDUS_STABLE_VERSIO` twice. cc @imeoer .

**Review by imeoer — APPROVED** [review]
Thanks!

**Pull request merged**